src/main/java/com/spotify/futures/FuturesExtra.java

```diff
@@ -172,47 +172,60 @@ public void onFailure(Throwable t) {
   }
 
   /**
-   * A lambda-friendly way to attach a callback to a {@link ListenableFuture}. The callback will
-   * only be run if the future completes successfully.
-   * @param future a ListenableFuture to attach the callback to.
-   * @param consumer a consumer, to be called with the result of the successful future.
+   * A lambda-friendly way to attach callbacks to a {@link ListenableFuture}. The success
+   * callback will only be run if the future completes successfully, and failure will
+   * only be run if the future fails.
+   * @param future a ListenableFuture to attach the callbacks to.
+   * @param success a consumer, to be called with the result of the successful future.
+   * @param failure a consumer, to be called with the result of the failed future.
+   * @throws NullPointerException if the {@param success} and {@param failure} are null
```

Comment:
this is slightly unclear (not obvious that both have to be null for it to throw) but i dont want to reject the change based on that.